erpnext/payroll/doctype/gratuity/gratuity.json

```diff
@@ -0,0 +1,201 @@
+{
+ "actions": [],
+ "autoname": "HR-GRA-PAY-.#####",
+ "creation": "2020-08-05 20:52:13.024683",
+ "doctype": "DocType",
+ "editable_grid": 1,
+ "engine": "InnoDB",
+ "field_order": [
+  "employee",
+  "employee_name",
+  "department",
+  "designation",
+  "column_break_3",
+  "posting_date",
+  "status",
+  "company",
+  "gratuity_rule",
+  "section_break_5",
+  "pay_via_salary_slip",
+  "payroll_date",
+  "salary_component",
+  "expense_account",
```

Comment:
Currently, you are using an expense account to map with party account in payment entry. It should not be like this.

On submission of Gratuity, we should accrue the expense. Add another field for getting "Payable Account" which will be a liability account.

Payable Account - Credit
Expense account - Debit

And in payment entry, payable account should be mapped with party account.

All this will be applicable and visible only if "pay_via_salary_slip" is unchecked. Otherwise, accrual will happen in payroll entry.